pyanaconda/ui/tui/spokes/__init__.py

```diff
@@ -175,9 +179,20 @@ def input(self, entry, key):
                 print(self.wrong_input_message)
             else:
                 print(_("You have provided an invalid user name: %s\n"
-                        "Tip: Keep your user name shorter than 32 characters and do not use spaces.\n") % key)
+                        "%s\n"
+                        "Tip: Keep your user name shorter than 32 characters and do not use spaces.\n")) % (key, self._error_mes(key, not_allowed_dirs))
             return NormalTUISpoke.input(self, entry, key)
 
+    def _error_mes(self, in_user, not_allowed_dirs):
+        if in_user in not_allowed_dirs:
+            return "This name is reserved for system"
+        elif re.search("^[\d]", in_user):
+            return "Username cannot start with digit"
+        elif re.search("[\D]", in_user):
+            return "Invalid character in username: %s" % re.search("[\D]", in_user).group()
```

Comment:
Are you sure about this check? AFAICT it means "if there's something else than digits in ``in_user``".